closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -135,15 +136,22 @@ private RestrictedSecurity() {
     }
 
     private static boolean isJarVerifierInStackTrace() {
-        java.util.function.Predicate<Class<?>> isJarVerifier =
-                clazz -> "java.util.jar.JarVerifier".equals(clazz.getName())
-                      && "java.base".equals(clazz.getModule().getName());
-
-        java.util.function.Function<Stream<StackWalker.StackFrame>, Boolean> matcher =
-                stream -> stream.map(StackWalker.StackFrame::getDeclaringClass)
-                                .anyMatch(isJarVerifier);
-
-        return StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).walk(matcher);
+        final String targetClass = "java.util.jar.JarVerifier";
+        final String targetModule = "java.base";
+
+        for (java.util.Map.Entry<Thread, StackTraceElement[]> e : Thread.getAllStackTraces().entrySet()) {
+            StackTraceElement[] stack = e.getValue();
+            if (stack == null)
+                continue;
```

Comment:
I think this part of the change should be proposed separately (see the second point in https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/1097#issuecomment-3303975488).

Reply:
Perhaps I was mistaken. Looking again, having any thread executing anything in `JarVerifier`, just delays the hash checking.

I still don't like the fact that it checks _all_ stack frames of _all_ threads, but I suppose there are _likely_ to be fewer threads before hash checking is complete, but it's not guaranteed.